packages/teko/CMakeLists.txt

```diff
@@ -14,47 +14,47 @@ TRIBITS_PACKAGE(Teko)
-      ENDIF () # Tpetra_INST_INT_INT
-    ELSE () # NOT DEFINED Tpetra_INST_INT_INT
-      MESSAGE (FATAL_ERROR "Teko currently requires that Tpetra enable either GlobalOrdinal (GO) = long long or GO = int.  Please set Tpetra_INST_INT_LONG_LONG=ON to enable GO = long long, or Tpetra_INST_INT_INT=ON to enable GO = int.  If you would like to use other GO types, please contact the Teko developers.")
-    ENDIF ()
-  ENDIF () # Tpetra_INST_INT_LONG_LONG
-ELSE () # NOT DEFINED Tpetra_INST_INT_LONG_LONG
-  IF (NOT DEFINED Tpetra_INST_INT_INT)
-    MESSAGE (FATAL_ERROR "Teko currently requires that Tpetra enable either GlobalOrdinal (GO) = long long or GO = int.  Please set Tpetra_INST_INT_LONG_LONG=ON to enable GO = long long, or Tpetra_INST_INT_INT=ON to enable GO = int.  If you would like to use other GO types, please contact the Teko developers.")
-  ELSEIF (NOT Tpetra_INST_INT_INT)
-    MESSAGE (FATAL_ERROR "Teko currently requires that Tpetra enable either GlobalOrdinal (GO) = long long or GO = int.  Please set Tpetra_INST_INT_LONG_LONG=ON to enable GO = long long, or Tpetra_INST_INT_INT=ON to enable GO = int.  If you would like to use other GO types, please contact the Teko developers.")
-  ENDIF ()
-  SET(${PACKAGE_NAME}_USE_LONGLONG_GO_DEFAULT OFF)
-ENDIF ()
-
-ASSERT_DEFINED(${PACKAGE_NAME}_USE_LONGLONG_GO_DEFAULT)
-TRIBITS_ADD_OPTION_AND_DEFINE(
-  ${PACKAGE_NAME}_USE_LONGLONG_GO
-  HAVE_${PACKAGE_NAME}_USE_LONGLONG_GO
-  "Use long long for global ordinal"
-  ${${PACKAGE_NAME}_USE_LONGLONG_GO_DEFAULT}
-  )
-
-ASSERT_DEFINED(${PACKAGE_NAME}_USE_LONGLONG_GO)
-IF (${PACKAGE_NAME}_USE_LONGLONG_GO)
-  IF (DEFINED Tpetra_INST_INT_LONG_LONG)
-    IF (NOT Tpetra_INST_INT_LONG_LONG)
-      MESSAGE(FATAL_ERROR "Teko: If GO=long long is disabled in Tpetra, you may not set ${PACKAGE_NAME}_USE_LONGLONG_GO = ON.  First, set Tpetra_INST_INT_LONG_LONG=ON to enable GO=long long in Tpetra.")
-    ENDIF ()
-  ELSE () # NOT DEFINED Tpetra_INST_INT_LONG_LONG
-    MESSAGE(FATAL_ERROR "Teko: If GO=long long is disabled in Tpetra, you may not set ${PACKAGE_NAME}_USE_LONGLONG_GO = ON.  First, set Tpetra_INST_INT_LONG_LONG=ON to enable GO=long long in Tpetra.")      
-  ENDIF ()
-ENDIF ()
+#IF (DEFINED Tpetra_INST_INT_LONG_LONG)
+#  IF (Tpetra_INST_INT_LONG_LONG)
+#    SET(${PACKAGE_NAME}_USE_LONGLONG_GO_DEFAULT ON)
+#  ELSE ()
+#    IF (DEFINED Tpetra_INST_INT_INT)
+#      IF (Tpetra_INST_INT_INT)
+#	SET(${PACKAGE_NAME}_USE_LONGLONG_GO_DEFAULT OFF)
+#      ELSE ()
+#	MESSAGE (FATAL_ERROR "Teko currently requires that Tpetra enable either GlobalOrdinal (GO) = long long or GO = int.  Please set Tpetra_INST_INT_LONG_LONG=ON to enable GO = long long, or Tpetra_INST_INT_INT=ON to enable GO = int.  If you would like to use other GO types, please contact the Teko developers.")
```

Comment:
We'll have to ask a @trilinos/teko developer whether Teko needs those defines, in case `Tpetra_INST_INT_LONG_LONG` or `Tpetra_INST_INT_INT` are set.  On the other hand, we plan to land the big "remove deprecated code" PR #6388 tomorrow.  That PR would make it so that Tpetra only ever has one `GO` type enabled.  Thus, it should be correct for Teko not to need these CMake options any more; in fact, there would be no way to use them once deprecated code is removed.

Reply:
I agree - I had the same concern.  @eric-c-cyr , can you please have a look and comment?

Reply:
I think we resolved this in our email chain with @eric-c-cyr ...